plugin/innodb_memcached/daemon_memcached/daemon/memcached.c

```diff
@@ -4571,7 +4570,9 @@ static char* process_command(conn *c, char *command) {
     }
 
     ntokens = tokenize_command(command, tokens, MAX_TOKENS);
-    if (ntokens >= 3 &&
+    if (ntokens > MAX_TOKENS) { /* fixes issues with "gcc-7 -O3" */
```

Comment:
What was the problem here?

Reply:
```
In function â€˜detokenize.constpropâ€™,
    inlined from â€˜process_statâ€™ at /github/inikep-ps/plugin/innodb_memcached/daemon_memcached/daemon/memcached.c:3962:13,
    inlined from â€˜process_commandâ€™ at /github/inikep-ps/plugin/innodb_memcached/daemon_memcached/daemon/memcached.c:4615:15,
    inlined from â€˜try_read_commandâ€™ at /github/inikep-ps/plugin/innodb_memcached/daemon_memcached/daemon/memcached.c:4867:16:
/github/inikep-ps/plugin/innodb_memcached/daemon_memcached/daemon/memcached.c:3594:11: error: argument 1 range [18446744071562067968, 18446744073709551615] exceeds maximum object size 9223372036854775807 [-Werror=alloc-size-larger-than=]
     buf = malloc(nb * sizeof(char));
           ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/x86_64-linux-gnu/bits/string2.h:1273,
                 from /usr/include/string.h:630,
                 from /usr/include/x86_64-linux-gnu/sys/un.h:37,
                 from /github/inikep-ps/plugin/innodb_memcached/daemon_memcached/config_static.h:84,
                 from /github/inikep-ps/plugin/innodb_memcached/daemon_memcached/daemon/memcached.c:20:
/github/inikep-ps/plugin/innodb_memcached/daemon_memcached/daemon/memcached.c: In function â€˜try_read_commandâ€™:
/usr/include/stdlib.h:466:14: note: in a call to allocation function â€˜mallocâ€™ declared here
 extern void *malloc (size_t __size) __THROW __attribute_malloc__ __wur;
              ^~~~~~
```

Reply:
It concerns:
```
    } else if (ntokens >= 2 && (strcmp(tokens[COMMAND_TOKEN].value, "stats") == 0)) {

        ret = process_stat(c, tokens, ntokens);

    } 
```

Reply:
It looks to me as a `gcc` optimization bug. `gcc` claims that `ntokens` can be a very big number, what is possible if `ntokens < 2` but this is checked.
`ntokens = tokenize_command(command, tokens, MAX_TOKENS);` should return `MAX_TOKENS` as maximum.